v25.2 GA (#19597)
cockroachdb/docs: src/current/_includes/releases/v25.2/v25.2.0.md

    </p>
   </td>
   <td>25.2</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
  </tr>
 </tbody>
</table>

<h4 id="v25-2-0-change-data-capture">Change Data Capture</h4>

<table>
 <thead>
  <tr>
   <th class="center-align" colspan="1" rowspan="2">Feature</th>
   <th class="center-align" colspan="5" rowspan="1">Availability</th>
  </tr>
  <tr>
    <th colspan="1" rowspan="1">Ver.</th>
    <th colspan="1" rowspan="1" style="white-space: nowrap;">Self-hosted</th>
    <th colspan="1" rowspan="1">Advanced</th>
    <th colspan="1" rowspan="1">Standard</th>
    <th colspan="1" rowspan="1">Basic</th>
  </tr>
 </thead>
 <tbody>
  <tr>
   <td>

>     <p class="feature-summary">Major Reduction in Duplicate Messages for Changefeeds</p>

Reviewer: ```suggestion
    <p class="feature-summary">Major reduction in duplicate messages for changefeeds</p>
```

    <p class="feature-description">

>       We’ve introduced a major improvement to changefeeds that significantly reduces the number of <a href="https://www.cockroachlabs.com/docs/v25.2/changefeed-messages#duplicate-messages">duplicate messages</a> emitted during restarts and retries. This means more accurate data delivery, lower processing overhead for downstream consumers, and a smoother experience overall. This improvement is especially beneficial in systems with large tables or uneven processing speeds, where legacy checkpointing often led to unnecessary message duplication. With this update, your changefeeds are more reliable, efficient, and production-ready than ever.  

Reviewer: ```suggestion
      A major improvement to changefeeds significantly reduces the number of <a href="https://www.cockroachlabs.com/docs/v25.2/changefeed-messages#duplicate-messages">duplicate messages</a> emitted during restarts and retries. This means more accurate data delivery, lower processing overhead for downstream consumers, and a smoother experience overall. This improvement is especially beneficial in systems with large tables or uneven processing speeds, where legacy checkpointing often led to unnecessary message duplication.
```

    </p>
   </td>
   <td>25.2</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
  </tr>
  <tr>
   <td>

>     <p class="feature-summary">Enriched Changefeed format for easier integration and richer metadata</p>

Reviewer: ```suggestion
    <p class="feature-summary">Enriched changefeed format for easier integration and richer metadata</p>
```

    <p class="feature-description">

>       In 25.2 we’re introducing a preview of an enriched changefeed format that aligns with the Debezium message structure, making it significantly easier for to integrate CockroachDB into your existing CDC pipelines. This new format adds structured metadata and flexibility to the changefeed message body, allowing teams to seamlessly migrate from PostgreSQL or Debezium-based systems without re-architecting downstream services. Whether you're working with real-time analytics, event-driven systems, or audit logs, the enriched format provides clear context about each change event—including source information, schema, operation type, and timestamps. It simplifies debugging, enhances observability, and makes changefeeds more useful in complex, distributed architectures. This new metadata envelope is in preview in 25.2.  

Reviewer: ```suggestion
      An enriched changefeed format has been introduced in Preview that aligns with the Debezium message structure, making it significantly easier to integrate CockroachDB into existing CDC pipelines. This new format adds structured metadata and flexibility to the changefeed message body, allowing seamless migrations from PostgreSQL or Debezium-based systems without re-architecting downstream services. Whether you're working with real-time analytics, event-driven systems, or audit logs, the enriched format provides clear context about each change event, including source information, schema, operation type, and timestamps. It simplifies debugging, enhances observability, and makes changefeeds more useful in complex, distributed architectures.
```

    <th colspan="1" rowspan="1">Advanced</th>
    <th colspan="1" rowspan="1">Standard</th>
    <th colspan="1" rowspan="1">Basic</th>
  </tr>
 </thead>
 <tbody>
  <tr>
   <td>
    <p class="feature-summary">Configurable TLS cipher suite restrictions</p>
    <p class="feature-description">
      A new <code>cockroach start</code> flag, <code>--tls-cipher-suites</code>, allows administrators to restrict which <a href="https://www.cockroachlabs.com/docs/v25.2/authentication#supported-cipher-suites">supported TLS cipher suites</a> are permitted for incoming SQL, RPC, and HTTP connections. Connections using disallowed cipher suites are rejected and logged. The restriction applies to both TLS 1.2 and TLS 1.3, enabling tighter control over cryptographic standards in secure clusters.  
    </p>
   </td>
   <td>25.2</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
  </tr>
  <tr>
   <td>
    <p class="feature-summary">Row-Level Security</p>
    <p class="feature-description">

>       Controlling who can access what data is more important than ever as organizations scale and modernize their data infrastructure. As enterprises move to modernize their critical databases, they need fine-grained, built-in access controls that go beyond table-level permissions. That’s why, with the 25.2 release of CockroachDB, we’re introducing Row-Level Security — a powerful feature that allows you to define and enforce access policies at the row level, directly within the database. This form of mandatory access control enables developers and operators to tightly govern data visibility based on user roles or attributes — making it a natural fit for securing sensitive workloads and building robust multi-tenant applications. With row-level security, CockroachDB makes it simple to isolate data, comply with regulatory requirements, and reduce application-side complexity — all while maintaining performance at scale and with minimal application changes.  

Reviewer: ```suggestion
      Row-level security (RLS) has been introduced for enterprises that need fine-grained, built-in access controls that go beyond table-level permissions. RLS allows you to define and enforce access policies at the row level, directly within the database. This form of mandatory access control enables developers and operators to tightly govern data visibility based on user roles or attributes, and is well suited for securing sensitive workloads and building robust multi-tenant applications. Row-level security makes it simple to isolate data, comply with regulatory requirements, and reduce application-side complexity while maintaining performance at scale and with minimal application changes.  
```

    </p>
   </td>
   <td>25.2</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
  </tr>
 </tbody>
</table>

<h4 id="v25-2-0-sql">SQL</h4>

<table>
 <thead>
  <tr>
   <th class="center-align" colspan="1" rowspan="2">Feature</th>
   <th class="center-align" colspan="5" rowspan="1">Availability</th>
  </tr>
  <tr>
    <th colspan="1" rowspan="1">Ver.</th>
    <th colspan="1" rowspan="1" style="white-space: nowrap;">Self-hosted</th>
    <th colspan="1" rowspan="1">Advanced</th>
    <th colspan="1" rowspan="1">Standard</th>
    <th colspan="1" rowspan="1">Basic</th>
  </tr>
 </thead>
 <tbody>
  <tr>
   <td>
    <p class="feature-summary">Faster index creation on large tables</p>
    <p class="feature-description">

>       Schema change operations such as index creations are now up to 30% faster based on our testing with a sample workload on a 1TB table.  

Reviewer: ```suggestion
      Schema change operations such as index creations are now up to 30% faster, based on internal testing with a sample workload on a 1TB table.  
```

    </p>
   </td>
   <td>25.2</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
  </tr>
  <tr>
   <td>
    <p class="feature-summary">Vector Indexing available in Preview</p>
    <p class="feature-description">

>       CockroachDB now offers <a href="https://www.cockroachlabs.com/docs/v25.2/vector-indexes">vector indexing</a> in preview, enabling efficient storage and similarity search over high-dimensional data. This feature simplifies building recommendation engines, semantic search, and other machine-learning workflows directly in the database—letting you prototype and integrate advanced vector-based applications without external tooling while taking advantage of powerful SQL semantics.

Reviewer: ```suggestion
      CockroachDB now offers <a href="https://www.cockroachlabs.com/docs/v25.2/vector-indexes">vector indexing</a> in Preview, enabling efficient storage and similarity search over high-dimensional data. This feature simplifies building recommendation engines, semantic search, and other machine-learning workflows directly in the database.
```

    </p>
   </td>
   <td>25.2</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
  </tr>
  <tr>
   <td>
    <p class="feature-summary">JSONPath support in Preview</p>
    <p class="feature-description">

>       Preview support for <a href="https://www.cockroachlabs.com/docs/v25.2/jsonpath">JSONPath</a> enables powerful, declarative querying and manipulation of deeply nested JSON structures—making it easier to work with complex JSON data and prototype advanced JSON-based workflows.

Reviewer: ```suggestion
      Preview support for <a href="https://www.cockroachlabs.com/docs/v25.2/jsonpath">JSONPath</a> enables powerful, declarative querying and manipulation of deeply nested JSON structures. This makes it easier to work with complex JSON data and prototype advanced JSON-based workflows.
```

    </p>
   </td>
   <td>25.2</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
  </tr>
 </tbody>
</table>

<h4 id="v25-2-0-licensing">Licensing</h4>

<table>
 <thead>
  <tr>
   <th class="center-align" colspan="1" rowspan="2">Feature</th>
   <th class="center-align" colspan="5" rowspan="1">Availability</th>
  </tr>
  <tr>
    <th colspan="1" rowspan="1">Ver.</th>
    <th colspan="1" rowspan="1" style="white-space: nowrap;">Self-hosted</th>
    <th colspan="1" rowspan="1">Advanced</th>
    <th colspan="1" rowspan="1">Standard</th>
    <th colspan="1" rowspan="1">Basic</th>
  </tr>
 </thead>
 <tbody>
  <tr>
   <td>
    <p class="feature-summary">Manage Enterprise license keys in the Cloud Console</p>
    <p class="feature-description">

>       You can now manage new <a href="https://www.cockroachlabs.com/docs/v25.2/licensing-faqs">Enterprise license keys</a> for CockroachDB Self-Hosted in the CockroachDB Cloud Console. This feature can help you keep better track of your license keys, their expiration dates, and important notifications.

Reviewer: Use liquid var for Cloud here?

Author: Noted for follow-up

>       You can now manage new <a href="https://www.cockroachlabs.com/docs/v25.2/licensing-faqs">Enterprise license keys</a> for CockroachDB Self-Hosted in the CockroachDB Cloud Console. This feature can help you keep better track of your license keys, their expiration dates, and important notifications.

Reviewer: ```suggestion
      You can now manage new <a href="https://www.cockroachlabs.com/docs/v25.2/licensing-faqs">Enterprise license keys</a> for self-hosted CockroachDB clusters in the CockroachDB Cloud Console. This feature can help you keep better track of your license keys, their expiration dates, and important notifications.
```
